From 2612211aae0d3ac2f6ce7048e14ba6e3b429582a Mon Sep 17 00:00:00 2001
From: Fei Jiang <fei.jiang@intel.com>
Date: Fri, 5 Jan 2018 00:09:28 +0800
Subject: [PATCH 0495/1240] Revert "drm/i915/gvt: control the scheduler by
 timeslice usage"

This reverts commit b35f34d1da4e77637869c8041a355da810f69fb6.

Signed-off-by: Fei Jiang <fei.jiang@intel.com>

Conflicts:
	drivers/gpu/drm/i915/gvt/sched_policy.c

Change-Id: Icf7e6bff09db0c9a5afe0d0725c65ff5b73eee16
Reviewed-by: Jiang, Fei <fei.jiang@intel.com>
Reviewed-by: He, Min <min.he@intel.com>
Reviewed-by: Dong, Eddie <eddie.dong@intel.com>
Tested-by: Dong, Eddie <eddie.dong@intel.com>
---
 drivers/gpu/drm/i915/gvt/sched_policy.c | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/drivers/gpu/drm/i915/gvt/sched_policy.c b/drivers/gpu/drm/i915/gvt/sched_policy.c
index 6496857..6d6d480 100644
--- a/drivers/gpu/drm/i915/gvt/sched_policy.c
+++ b/drivers/gpu/drm/i915/gvt/sched_policy.c
@@ -183,11 +183,8 @@ static struct intel_vgpu *find_busy_vgpu(struct gvt_sched_data *sched_data)
 		if (!vgpu_has_pending_workload(vgpu_data->vgpu))
 			continue;
 
-		/* Return the vGPU only if it has time slice left */
-		if (vgpu_data->left_ts > 0) {
-			vgpu = vgpu_data->vgpu;
-			break;
-		}
+		vgpu = vgpu_data->vgpu;
+		break;
 	}
 
 	return vgpu;
@@ -215,8 +212,8 @@ static void tbs_sched_func(struct gvt_sched_data *sched_data)
 		list_del_init(&vgpu_data->lru_list);
 		list_add_tail(&vgpu_data->lru_list,
 				&sched_data->lru_runq_head);
-	} else {
-		scheduler->next_vgpu = gvt->idle_vgpu;
+
+		gvt_dbg_sched("pick next vgpu %d\n", vgpu->id);
 	}
 out:
 	if (scheduler->next_vgpu)
-- 
2.7.4

